### PR TITLE
Wire up Behemoth Suit +1

### PR DESCRIPTION
**Pull request by neuromancerxi**
Wires up Behemoth Suit +1

Data pulled from 164 uses across 12 characters during Return Home Campaign 2022. While community sources claim to have Savory Shank as a potential drop, I was unable to prove this out. Added commented out Savory Shank in the event that additional data is later able to prove this out.

Item Frequency Data:

Buffalo Meat: 9
Cerberus Meat: 4
Cockatrice Meat: 17
Coeurl meat: 13
Dhalmel Meat: 20
Dragon Meat: 5
Giant Sheep Meat: 17
Hare Meat: 22
Hydra Meat: 5
Karakul Meat: 4
Land Crab Meat: 9
Lucerewe Meat: 7
Lynx Meat: 5
Orobon Meat: 9
Ruszor Meat: 9
Ziz Meat: 7
Beastly Shank: 2

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Wires up Behemoth Suit +1 as usable and provides supporting script to provide a weighted random item from observed pool of results.

## Steps to test these changes

`!additem 26955`
Equip Suit
Use Item, get meat!

For the purposes of testing, testers may want to lower the reuse timer below `72000` to test sufficient randomization of resulting meat.

### CODE REVIEW

File: scripts/globals/items/behemoth_suit_+1.lua [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    return result
+end
+
+item_object.onItemUse = function(target)
+    local meatList = 
+            {
+                { 9, xi.items.SLICE_OF_BUFFALO_MEAT}, -- 5152
+                { 4, xi.items.SLICE_OF_CERBERUS_MEAT}, -- 5565
+                {17, xi.items.SLICE_OF_COCKATRICE_MEAT}, -- 4435
+                {13, xi.items.SLICE_OF_COEURL_MEAT}, -- 4377
```
Comment (Xaver-DaRed) [inline scripts/globals/items/behemoth_suit_+1.lua]: Style:
```
    local meatList = 
    {
        { 9, xi.items.SLICE_OF_BUFFALO_MEAT}, -- 5152
        { 4, xi.items.SLICE_OF_CERBERUS_MEAT}, -- 5565
        {17, xi.items.SLICE_OF_COCKATRICE_MEAT}, -- 4435
        {13, xi.items.SLICE_OF_COEURL_MEAT}, -- 4377
```
And so on.

The rest looks good^^

Comment (neuromancerxi) [reply scripts/globals/items/behemoth_suit_+1.lua]: Fixed! Appreciate the feedback!

**Comment by TeoTwawki** [comment]
hey thanks for doing this.

> Added commented out Savory Shank in the event that additional data is later able to prove this out.

good call :+1: it probably is there at the same infinitesimally mall rate as the speed belt from arthro hat +1 which takes about 2000 uses to get one @_@

File: scripts/globals/item_utils.lua [hunk 2]
```diff
@@ -41,3 +41,28 @@ end
 xi.item_utils.skillBookUse = function(target, skillID)
     target:trySkillUp(skillID, target:getMainLvl(), true, true)
 end
+
+xi.item_utils.pickItemRandom = function(target, itemgroup) -- selects an item from a weighted result table
```
Comment (TeoTwawki) [inline scripts/globals/item_utils.lua:45]: _Shockingly_ we didn't already have this. Will likely expand this out to multiple other places where we basically do the exact same thing.

**Review by claywar — APPROVED** [review]

**Pull request merged**